### PR TITLE
Migrate Models Variable to Internal API

### LINKED ISSUE #28271: AIP-44 Migrate Variable to Internal API
Link: https://github.com/apache/airflow/blob/main/airflow/models/variable.py

Methods to migrate:

- val
- set 
- delete
- update

Note that get_variable_from_secrets shouls still be executed locally.

It may be better to first close https://github.com/apache/airflow/issues/28267

### PR DESCRIPTION
**Pull request by bugraoz93**
Migration of following methods of `Models.Variable`  to Internal API. 
- set
- update
- delete

Closes: #28271

### CODE REVIEW

File: airflow/models/variable.py
Comment (potiuk) [inline airflow/models/variable.py]: Why not @staticmethod here (As opposed to set above)?

Comment (bugraoz93) [reply airflow/models/variable.py]: My bad. Thanks for pointing it out! It should be **@staticmethod**. I changed it to **@staticmethod**.

Comment (potiuk) [reply airflow/models/variable.py]: Then also `cls` has to be removed.

Comment (bugraoz93) [reply airflow/models/variable.py]: Definitely :)

**Review by potiuk — APPROVED** [review]
LGTM pending tests succeed.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Needs conflict resolution after other merges.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
And yeah - re-rebasing (confllicts resolution) is needed

**Comment by bugraoz93** [comment]
You are right, resolved.

**Pull request merged**
